src/components/TextArea.js

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useState } from 'react'
+import styled from 'styled-components'
+import { TextInputLike } from '../components/Common.js'
+
+const TextAreaContainer = styled.div`
+  margin: 1em;
+`
+
+const TextAreaBox = styled.textarea.attrs({
+  type: 'text',
+})`
+  height: 200px;
+  width: 600px;
+  ${TextInputLike};
+  ::-webkit-scrollbar {
+    width: 10px;
+    height: 10px;
+  }
+  ::-webkit-scrollbar-thumb {
+    background-color: transparent;
+    border-radius: 10px;
+    border: 1px solid ${p => p.theme.colors.highlight};
+  }
+  ::-webkit-scrollbar-track {
+    background-color: transparent;
+  }
+  ::-webkit-scrollbar-corner {
+    background-color: transparent;
+  }
+  ::-webkit-resizer {
+    background-color: transparent;
+  }
+`
+const ErrorMsg = styled.p`
+  color: ${p => p.theme.colors.warning};
+  margin: 7px 0px;
+`
+
+export const TextArea = ({
+  value,
+  onChangeValue,
+  maxLength,
+  placeholder,
+  invalid,
+  errorMsg,
+  ...rest
+}) => {
+  const [isLengthExceeded, setIsLengthExceeded] = useState(false)
+  useEffect(() => {
+    // debounced with setTimeout
+    const handler = setTimeout(() => {
+      setIsLengthExceeded(maxLength && value && value.length > maxLength)
+    }, 500)
+
+    return () => {
+      clearTimeout(handler)
+    }
+  }, [maxLength, value])
+
+  return (
+    <TextAreaContainer>
+      <TextAreaBox
+        value={value}
+        onChange={val => {
+          onChangeValue(val.target.value)
+        }}
```

Comment:
Just to keep the `onChange` handler consistent with all of the other input components that have been built (text, checkbox, radio, etc), can we change this to 
```suggestion
        onChange={onChange}
```
and rename `onChangeValue` to `onChange` in line 41?

Reply:
thanks for pointing this out! will follow up after I make a change (that's not exactly the same as this suggestion)